compiler/src/dotty/tools/dotc/typer/ProtoTypes.scala

```diff
@@ -71,6 +71,10 @@ object ProtoTypes {
                    |constraint was: ${ctx.typerState.constraint}
                    |constraint now: ${newctx.typerState.constraint}""")
             if result && (ctx.typerState.constraint ne newctx.typerState.constraint) then
+              val tvars = (newctx.typerState.ownedVars -- ctx.typerState.ownedVars).toList
+              inContext(newctx):
+                Inferencing.instantiateSelected(tp, tvars)
```

Comment:
If the tvars are created from the TypeLambda underlying `tp`, then I expect that they wouldn't be present at all in `tp` so this line would be a no-op, is that not the case?

Reply:
I'll check

Reply:
You're right.  Wasn't looking at the right type.